core/src/main/java/org/infinispan/distribution/DistributionInfo.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.distribution;
+
+import java.util.List;
+
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.remoting.transport.Address;
+
+public class DistributionInfo {
```

Comment:
There is only one place with a switch-like check. I wanted to compute as little as possible, though I understand your desire; I can replace `isPrimary()` and `isBackup()` with an enum...